modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx

```diff
@@ -548,5 +592,66 @@ export function ModalAddFilter({
 				}
 			/>
 		</ClayModal>
-	);
+	) : null;
 }
+
+export default ModalAddFilter;
+
+interface IProps {
+	currentFilters: CurrentFilter[];
+	disableDateValues?: boolean;
+	editingObjectFieldName: string;
+	filterEndDate: string;
+	filterOperators: TFilterOperators;
+	filterStartDate: string;
+	filterTypeRequired?: boolean;
+	header?: string;
+	modalType: string;
+	objectFields: ObjectField[];
+	onSave?: (
+		objectFieldName: string,
+		filterBy?: string,
+		fieldLabel?: LocalizedValue<string>,
+		objectFieldBusinessType?: string,
+		filterType?: string,
+		valueList?: IItem[],
+		value?: string
+	) => void;
+	selectedFilterBy?: ObjectField | null;
+	selectedFilterType?: LabelValueObject | null;
+	showModal: boolean;
+	validate?: ({
+		checkedItems,
+		disableDateValues,
+		items,
+		selectedFilterBy,
+		selectedFilterType,
+		setErrors,
+		value,
+	}: FilterValidation) => FilterErrors;
+	value?: string;
+	workflowStatusJSONArray: LabelValueObject[];
+}
+
+interface IItem extends LabelValueObject {
+	checked?: boolean;
+}
+
+export type FilterErrors = {
+	endDate?: string;
+	items?: string;
+	selectedFilterBy?: string;
+	selectedFilterType?: string;
+	startDate?: string;
+	value?: string;
+};
+
+export type FilterValidation = {
+	checkedItems: IItem[];
+	disableDateValues?: boolean;
+	items: IItem[];
+	selectedFilterBy?: ObjectField | null;
+	selectedFilterType?: LabelValueObject | null;
+	setErrors: (value: FilterErrors) => void;
+	value?: string;
+};
```

Comment:
@rebsilva, could you move all these types and interfaces to the top of the file?